### PR TITLE
Added more to docstring of datetime.timedelta

### PR DESCRIPTION
**Pull request by ChrisBarker-NOAA**
Addition of signature and a bit more text to the class docstring for datetime.datetime, as discussed on python-dev 5 June, 2018.

https://mail.python.org/pipermail/python-dev/2018-June/153813.html

NOTE: it would be nice to get this back-ported to all actively maintained versions, but I have no idea how to trigger that.

But if this is approved, and someone lets me know how to do it, I'll be happy to do so.

Also -- I couldn't find clear guidance in PEP 257 for docstrings for C classes, so I hope this is close.

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](https://bugs.python.org/) (b.p.o) to verify you have signed the CLA (this might be simply due to a missing "GitHub Name" entry in your b.p.o account settings). This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


When your account is ready, please add a comment in this pull request
and a Python core developer will remove the `CLA not signed` label
to make the bot check again.

Thanks again to your contribution and we look forward to looking at it!

**Review by abalkin — APPROVED** [review]

**Comment by ChrisBarker-NOAA** [comment]
thanks!

**Comment by vstinner** [comment]
Thanks @ChrisBarker-NOAA! I checked help(datetime.timedelta) before merging, and the docstring is properly rendered. It's a nice enhancement, I merged it.

> NOTE: it would be nice to get this back-ported to all actively maintained versions, but I have no idea how to trigger that.

I backported the change to Python 3.6 and 3.7. We have a bot doing the job for us!

(There is a conflict on Python 2.7, git cherry-pick fails to automatically backports the change. Honestly, I don't think that it's worth it to backport it up to Python 2.7.)

**Comment by ChrisBarker-NOAA** [comment]
Thanks — and I agree. Maybe one more reason to move off py2 :-)

**Comment by vstinner** [comment]
> I agree. Maybe one more reason to move off py2 :-)

LOL, yes, exactly!

**Pull request merged**
